docs/cli_hacking_guide.adoc

```diff
@@ -0,0 +1,293 @@
+= OpenShift 3 Command Line Interface (CLI)
```

Comment:
Both of our cli docs have the same header, should we change this to **CLI Hacking Guide** and remove the introduction about the cli (we already introduce it in cli.md) ?